packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -50,55 +36,47 @@
       limit: 100,
     },
   })
-  $: hasFilter = !!filter?.filter(f => !!f.field)?.length
-  $: fetch.update({ filter })
-  $: {
-    options = searchResults ? searchResults : $fetch.rows
-    const nonMatchingOptions = selectedOptions.filter(
-      option => !options.map(opt => opt._id).includes(option._id)
-    )
-    // Append initially selected options if there is no filter
-    // and hasn't already been appended
-    if (!hasFilter) {
-      options = [...options, ...nonMatchingOptions]
-    }
-  }
+
   $: tableDefinition = $fetch.definition
-  $: primaryDisplay = tableDefinition?.primaryDisplay || "_id"
-  $: singleValue = flatten(fieldState?.value)?.[0]
-  $: multiValue = flatten(fieldState?.value) ?? []
+  $: selectedValue = multiselect
+    ? flatten(fieldState?.value) ?? []
+    : flatten(fieldState?.value)?.[0]
   $: component = multiselect ? CoreMultiselect : CoreSelect
   $: expandedDefaultValue = expand(defaultValue)
-  $: debouncedSearch(searchString)
-  $: {
-    if (searching) {
-      isOpen = true
-    }
+  $: primaryDisplay = tableDefinition?.primaryDisplay
+
+  $: initialValues =
+    initialValues ||
+    (primaryDisplay &&
+      fieldState?.value?.map(x => ({
+        _id: x._id,
+        [primaryDisplay]: x.primaryDisplay,
+      })))
+
+  $: allOptions = {
+    ...(allOptions || {}),
+    ...[...($fetch.rows || []), ...(initialValues || [])]?.reduce((p, c) => {
```

Comment:
So many spread operators! Is there a way to make this a bit more readable? Are all these spreads necessary?

Same comment as before about what are 'p' and 'c' ?